### PR TITLE
Finished lesson_4.1

### PR DESCRIPTION
**Pull request by jennycodingnow**
Finished Lesson_4_1 converting to Typescript

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,17 @@ dist-ssr
...
+
+
+# Compiled JavaScript files
+*.js
+
+# TypeScript declaration files
+*.d.ts
+
+# TypeScript configuration files
+tsconfig.json
```
Comment (ElijahMSmith) [inline .gitignore]: These configuration files need to be committed! They tell the TypeScript compiler what rules they need to enforce on your TypeScript files, and without them my IDE is blowing up with red squiggles.

Comment (ElijahMSmith) [reply .gitignore]: Specifically `tsconfig.json` and `tsconfig.node.json`. I think `vite.config.json` should also be committed. Generally these config files should be shared by your whole team so everyone's IDE works the same.

File: src/App.test.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { render, screen, act } from "@testing-library/react";
 import { describe, it, expect, vitest } from "vitest";
-import App from "./App.jsx";
+import App from "./App.js";
```
Comment (ElijahMSmith) [inline src/App.test.tsx]: I think this import is going to need to be `import App from "./App";`. If you remove the import, then delete and re-type one of the `<App/>` tags below it should fix it for you.

**Review by ElijahMSmith — APPROVED** [review]
Great work!

File: src/App.test.tsx (hunk 2 above)
Comment (jennycodingnow) [reply src/App.test.tsx]: It didn't work. I tried your suggestions.

**Pull request merged**